src/modules/ipfix/IpfixPrinter.cpp

```diff
@@ -603,7 +603,7 @@ void IpfixPrinter::printOneLineRecord(IpfixDataRecord* record)
 		if (fi != NULL && fi->type.length==2) {
 			srcport = ntohs(*reinterpret_cast<uint16_t*>(record->data+fi->offset));
 		}
-		snprintf(buf, ARRAY_SIZE(buf), "%hhu.%hhu.%hhu.%hhu:%hu", (srcip>>0)&0xFF, (srcip>>8)&0xFF, (srcip>>16)&0xFF, (srcip>>24)&0xFF, srcport);
+		snprintf(buf, ARRAY_SIZE(buf), "%hhu.%hhu.%hhu.%hhu:%hu", (uint8_t)((srcip>>0)&0xFF), (uint8_t)((srcip>>8)&0xFF), (uint8_t)((srcip>>16)&0xFF), (uint8_t)((srcip>>24)&0xFF), srcport);
```

Comment:
Would it perhaps be better to use PRIu8 as the format specifier here?

Reply:
Yep, this should be changed :)